orderbook/src/storage/memory.rs

```diff
@@ -115,6 +98,10 @@ impl Storage for OrderBook {
                         .buy_token
                         .map(|token| token == order.order_creation.buy_token)
                         .unwrap_or(true)
+                    && filter
+                        .uid
+                        .map(|uid| uid == order.order_meta_data.uid)
+                        .unwrap_or(true)
```

Comment:
This seems like a bug that existed before but is only now caught by the e2e test.